programs/gzip.c

```diff
@@ -416,6 +424,9 @@ decompress_file(struct libdeflate_decompressor *decompressor, const tchar *path,
 	if (ret != 0)
 		goto out_close_in;
 
+	if (!options->test) { /// indicate empty output file name, mock use of stdout
```

Comment:
`options->to_stdout` is always set when `options->test`, which implies `newpath == NULL` already.  So this part is unnecessary.